drivers/espi/espi_mchp_xec_host_v2.c

```diff
@@ -181,30 +191,75 @@ static void kbc0_ibf_isr(const struct device *dev)
 	struct espi_xec_data *const data =
 		(struct espi_xec_data *const)dev->data;
 
-
+#ifdef CONFIG_ESPI_PERIPHERAL_KBC_IBF_EVT_DATA
+	/* Chrome solution */
+	struct espi_event evt = { ESPI_BUS_PERIPHERAL_NOTIFICATION,
+		ESPI_PERIPHERAL_8042_KBC, ESPI_PERIPHERAL_NODATA
+	};
+	struct espi_evt_data_kbc *kbc_evt =
+				(struct espi_evt_data_kbc *)&evt.evt_data;
+	/*
+	 * Indicates if the host sent a command or data.
+	 * 0 = data
+	 * 1 = Command.
+	 */
+	kbc_evt->type = kbc_hw->EC_KBC_STS & MCHP_KBC_STS_CD ? 1 : 0;
+	/* The data in KBC Input Buffer */
+	kbc_evt->data = kbc_hw->EC_DATA;
+	/* KBC Input Buffer Full event */
+	kbc_evt->evt = HOST_KBC_EVT_IBF;
+#else
+	/* Windows solution */
 	/* The high byte contains information from the host,
 	 * and the lower byte speficies if the host sent
 	 * a command or data. 1 = Command.
 	 */
-	uint32_t isr_data = ((kbc_hw->EC_DATA & 0xFF) << E8042_ISR_DATA_POS) |
-				((kbc_hw->EC_KBC_STS & MCHP_KBC_STS_CD) <<
-				 E8042_ISR_CMD_DATA_POS);
+	uint32_t isr_data = ((kbc_hw->EC_KBC_STS & MCHP_KBC_STS_CD) <<
+				E8042_ISR_CMD_DATA_POS);
+	isr_data |= ((kbc_hw->EC_DATA & 0xFF) << E8042_ISR_DATA_POS);
 
 	struct espi_event evt = {
 		.evt_type = ESPI_BUS_PERIPHERAL_NOTIFICATION,
 		.evt_details = ESPI_PERIPHERAL_8042_KBC,
 		.evt_data = isr_data
 	};
-
+#endif
 	espi_send_callbacks(&data->callbacks, dev, evt);
 
 	mchp_xec_ecia_info_girq_src_clr(xec_kbc0_cfg.ibf_ecia_info);
 }
 
 static void kbc0_obe_isr(const struct device *dev)
 {
+#ifdef CONFIG_ESPI_PERIPHERAL_KBC_OBE_CBK
+	/* Chrome solution */
+	struct espi_xec_data *const data =
+		(struct espi_xec_data *const)dev->data;
+
+	struct espi_event evt = { ESPI_BUS_PERIPHERAL_NOTIFICATION,
```

Comment:
Same here. consider making this a constant actually since it's used twice

Reply:
`evt` structure variable is updated (through `kbc_evt` pointer) so cannot make it a constant.